corgi/tasks/prod_defs.py

```diff
@@ -67,6 +69,87 @@ def update_products() -> None:
                 parse_product_version(pd_product_version, product, product_node)
 
 
+def _find_by_cpe(cpe_patterns: list[str]) -> list[str]:
+    """Given a list of CPE patterns find all the CPEs from ET Variants which match it"""
+    if not cpe_patterns:
+        return []
+    regex_query = Q()
+    for pattern in cpe_patterns:
+        regex = pattern.replace(".", "\\.").replace("*", ".*")
+        regex_query |= Q(cpe__regex=regex)
+    return list(
+        CollectorErrataProductVariant.objects.filter(live=True)
+        .filter(regex_query)
+        .values_list("cpe", flat=True)
+    )
+
+
+def _match_stream_version_to_cpe_version(
+    version_streams: QuerySet[ProductStream],
+    cpe_version: str,
+    cpe: str,
+    el_by_cpe: defaultdict[str, list],
+    cpes_by_stream: defaultdict[str, list],
+):
+    """Iterate product stream versions, looking for matches to the cpe_version passed in.
+    If a match is found, store it in cpes_by_streams dict along with any el suffixes."""
+    for version_stream in version_streams:
+        if version_stream.version.removesuffix(".z") == cpe_version:
```

Comment:
Instead of iterating over the entire queryset, can we just filter it and iterate over only the matching values, like below?
```suggestion
    cpe_version_with_z = f"{cpe_version}.z"
    for version_stream_name in version_streams.filter(models.Q(version=cpe_version) | models.Q(version=cpe_version_with_z)).values_list("name", flat=True):
```

Reply:
good idea.